### DISCUSSION TITLE
`load_contract` for testing detectors may not be ideal for long term + Pluasible fix

### DISCUSSION THREAD

**Discussion by TilakMaddy** [Cyfrin/aderyn#284] [Ideas]
Current behavior:

Input : `Path/to/out/xxx/yyy/zzz/Contract.json` in out directory

Processing:
1. Deserialize the json file
2. Looking for the project root folder by searching for `out`'s parent directory then reading the source files from `src` to plug into the AST

-----------

Problems with this approach

Once `forge` is exposed to a contract with pragma `0.8.25` it installs solc 0.8.25 and then from then on, it does not  generate 0.8.24 json files for existing smart contracts (of course because of the nature of floating pragmas)

This breaks the arguments in `load_contract` if the json file name ends with `XXX.0.8.24.json`, etc (NOTE: This is not the majority of cases)

So you have to change all the tests now : (

-----------------

Better approach ( I think )

The input for `load_contract` should be the path to the solidity file itself. 

Then we can go the relevant out folder and pick up the json (This is good enough for most detectors. The only exceptions are the ones where the detctors are pragma version sensitive of which there are only a few - it might even be better if we yank them away from the `out` directory and keep them somewhere else where it's permanent.

`out` folder should be always re-generate-able by any computer and we do not have to be worrying about breaking test

----------

**Comment by alexroan** [comment]
Agree!

This is a natural successor to the theme: "[Don't require or support _any_ framework, just use Solidity files](https://github.com/Cyfrin/aderyn/discussions/252#discussioncomment-9020239)"
